wayk_proto/src/message/now_messages/sharing.rs

```diff
@@ -0,0 +1,91 @@
+use crate::message::NowString256;
+use num_derive::FromPrimitive;
+use std::str::FromStr;
+
+#[derive(Encode, Decode, FromPrimitive, Debug, PartialEq, Clone, Copy)]
+#[repr(u8)]
+pub enum SharingMessageType {
+    Suspend = 0x01,
+    Resume = 0x02,
+}
+
+#[derive(Encode, Decode, Debug, Clone)]
+pub struct NowSharingSuspendMsg {
+    subtype: SharingMessageType,
+    flags: u8,
+    reserved: u16,
+    pub message: NowString256,
+}
+
+impl Default for NowSharingSuspendMsg {
+    fn default() -> Self {
+        NowSharingSuspendMsg {
+            subtype: SharingMessageType::Suspend,
+            flags: 0,
+            reserved: 0,
+            message: NowString256::from_str("").unwrap(),
+        }
+    }
+}
+
+impl NowSharingSuspendMsg {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn new_with_message(message: NowString256) -> Self {
+        let mut suspend_sharing = Self::default();
+        suspend_sharing.message = message;
+        suspend_sharing
```

Comment:
C'est OK, mais tu peux aussi écrire

```suggestion
        Self {
            message,
            ...Self::default(),
        }
```